### DISCUSSION TITLE
New traces not created in some cases (where I think they should)

### DISCUSSION THREAD

**Discussion by tobias-** [open-telemetry/opentelemetry-java-instrumentation#2011] [Q&A]
There are some traces that should be auto-created by instrumentation imho:

* Initialization (e.g. psv `main()` or when the context is initialized/set up by e.g. war/ear). It's not in any way compilcated to add a `@WithSpan` for the main(), but it would be nice if it was there from the start.
* Timer tasks (e.g. `(new Timer("heartBeat", true))..scheduleAtFixedRate(new HeartBeatRunnable(), freqMillis, freqMillis);` )
   Every time `HeartBeatRunnable().run()` is invoked, it should be in a new trace. I can't come up with any scenario where any `schedule` should be part of the trace that scheduled the execution.

**Comment by iNikem** [comment]
Nice ideas, @tobias- . Can you create two separate tasks for them?
